### PR TITLE
More scores

### PR DESCRIPTION
**Pull request by tjorim**
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: bot.user.js [hunk 1]
```diff
@@ -973,33 +1033,37 @@ window.sosBackup = sos;
...
+    userInterface.hideTop();
+
     // Pref display
     userInterface.onPrefChange();
 
+    // Modify the redraw()-function to remove the zoom altering code.
+    var original_redraw = window.redraw.toString();
+    var new_redraw = original_redraw.replace(
+        'gsc!=f&&(gsc<f?(gsc+=2E-4,gsc>=f&&(gsc=f)):(gsc-=2E-4,gsc<=f&&(gsc=f)))', '');
+    window.redraw = new Function(new_redraw.substring(
```
Comment (tjorim) [inline bot.user.js]: Not this again -_- (ignore)

File: bot.user.js [hunk 2]
```diff
@@ -434,37 +424,38 @@ var bot = (function() {
...
         stopBot: function() {
-            window.log('Stopping Bot.');
             window.setAcceleration(0); // Stop boosting
             // Re-enable the original onmousemove function
             window.onmousemove = original_onmousemove;
+            
+            window.log('Stopping Bot.');
             bot.isBotRunning = false;
+            // Update the HUD
+            userInterface.onPrefChange();
```
Comment (tjorim) [inline bot.user.js:447]: Ignore, just the arrangement of modules.

File: bot.user.js [hunk 3]
```diff
@@ -434,37 +424,38 @@ var bot = (function() {
         launchBot: function() {
             window.log('Starting Bot.');
             bot.isBotRunning = true;
+            // Update the HUD
+            userInterface.onPrefChange();
```
Comment (tjorim) [inline bot.user.js:426]: Ignore, just the arrangement of modules.

File: bot.user.js [hunk 4]
```diff
@@ -552,18 +542,18 @@ var bot = (function() {
...
+                };
+                canvasPosB = {
                     x: curpos.x + bot.heading.x*100,
                     y: curpos.y + bot.heading.y*100,
                     radius: 1
-                });
+                };
 
-                canvas.drawLine2(canvasPosA.x, canvasPosA.y, canvasPosB.x, canvasPosB.y, 2, 'yellow');
+                canvas.drawLine(canvasPosA, canvasPosB, 'yellow', 2);
```
Comment (tjorim) [inline bot.user.js:554]: Yes it is, a few lines above you.

File: bot.gridsystem.js [hunk 5]
```diff
@@ -593,18 +593,18 @@ g
...
+                    };
+                    var canvasPosB = {
                         x: snk.xx + aggressor.heading.x*100,
                         y: snk.yy + aggressor.heading.y*100,
                         radius: 1
-                    });
+                    };
 
-                    canvas.drawLine2(canvasPosA.x, canvasPosA.y, canvasPosB.x, canvasPosB.y, 2, 'yellow');
+                    canvas.drawLine(canvasPosA, canvasPosB, 'yellow', 2);
```
Comment (tjorim) [inline bot.gridsystem.js:608]: Ignore, canvas is a module in bot.user.js. Hound does not know this.

File: bot.user.js [hunk 6]
```diff
@@ -434,37 +422,38 @@ var bot = (function() {
...
         stopBot: function() {
-            window.log('Stopping Bot.');
             window.setAcceleration(0); // Stop boosting
             // Re-enable the original onmousemove function
             window.onmousemove = original_onmousemove;
+
+            window.log('Stopping Bot.');
             bot.isBotRunning = false;
+            // Update the HUD
+            userInterface.onPrefChange();
```
Comment (tjorim) [inline bot.user.js:447]: Ignore, just the arrangement of modules.

File: bot.user.js [hunk 7]
```diff
@@ -351,6 +351,14 @@ var canvas = (function() {
             return dir;
         },
 
+        // relativeAngle
+        getRelativeAngle: function(from, to) {
+            var norm = canvas.getNormalVector(from, to);
+            norm.dot = norm.x * bot.heading.x + norm.y * bot.heading.y;
```
Comment (tjorim) [inline bot.user.js:357]: Ignore, just the arrangement of modules.

File: bot.gridsystem.js [hunk 8]
```diff
@@ -99,21 +99,21 @@ var collisionHelper = (function() {
...
-                    });
-
-                    color = (!result.node||result.node.type==TYPE_EMPTY) ? 'green' : ((result.node.type==TYPE_FOOD) ? 'blue' : 'red');
-                    if( color != 'green')
-                    canvas.drawLine2(canvasPosA.x, canvasPosA.y, canvasPosB.x, canvasPosB.y, 1, color);
+                    }); */
+                    color = (!result.node||result.node.type==TYPE_EMPTY) ? 'green' :
+                        ((result.node.type==TYPE_FOOD) ? 'blue' : 'red');
+                    if( color != 'green' )
+                        canvas.drawLine(curpos, linePos, color, 1);
```
Comment (tjorim) [inline bot.gridsystem.js:116]: Ignore, canvas is a module in bot.user.js. Hound does not know this.

File: bot.behaviors.js [hunk 9]
```diff
@@ -223,15 +223,8 @@ var behaviors = (function() {
...
-                    x: window.snake.xx,
-                    y: window.snake.yy
-                };
-                canvas.drawLine(
-                    canvas.mapToCanvas(headCoord),
-                    canvas.mapToCanvas(bot.behaviorData.goalCoordinates),
-                    'green');
-
+                var headCoord = window.getPos();
+                canvas.drawLine(headCoord, bot.behaviorData.goalCoordinates, 'green');
```
Comment (tjorim) [inline bot.behaviors.js:227]: Ignore, canvas and bot are modules in bot.user.js. Hound does not know this.

File: bot.behaviors.js [hunk 10]
```diff
@@ -292,10 +285,8 @@ var behaviors = (function() {
                     x: obj.surroundExitPos.x,
                     y: obj.surroundExitPos.y,
                     radius: 1
-                });
-
-
-                canvas.drawLine2(canvasPosA.x, canvasPosA.y, canvasPosB.x, canvasPosB.y, 1, 'blue');
+                }); */
+                canvas.drawLine(curpos, obj.surroundExitPos, 'blue', 1);
```
Comment (tjorim) [inline bot.behaviors.js:289]: Ignore, canvas is a module in bot.user.js. Hound does not know this.

**Comment by ermiyaeskandary** [comment]
I think I can fix this module issue...

File: bot.user.js (hunk 4 above)
Comment (ermiyaeskandary) [inline bot.user.js:554]: lmao :+1:

**Comment by tyschnoor** [comment]
This is amazing!

**Comment by ermiyaeskandary** [comment]
Approved

[Approved with PullApprove](https://pullapprove.com/ErmiyaEskandary/Slither.io-bot/pull-request/203/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=ErmiyaEskandary/Slither.io-bot)

**Comment by Drflash55** [comment]
Lol, I love it how hound keeps finding errors that are blatantly not errors at all...

**Pull request merged**